app/src/androidTest/java/org/schabi/newpipe/local/playlist/LocalPlaylistManagerTest.kt

```diff
@@ -0,0 +1,41 @@
+package org.schabi.newpipe.local.playlist
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.schabi.newpipe.TestApp
+import org.schabi.newpipe.database.AppDatabase
+import org.schabi.newpipe.database.stream.model.StreamEntity
+import org.schabi.newpipe.extractor.stream.StreamType
+import javax.inject.Inject
+
+@RunWith(AndroidJUnit4::class)
+class LocalPlaylistManagerTest {
+    @Inject
+    lateinit var appDatabase: AppDatabase
```

Comment:
This is the actual implementation and not a mock/fake right?

How would i go about mocking this, so that i could convert it to an unit test, instead of instrumented. Same for when i want to run different `Mockito.verify()` or `Mockito.when()` depending on the test

Reply:
I created a `TestRoomModule` where the testing version of the database is created, so the actual database implementation can be replaced with a mock one, and Dagger will inject the mock version.

Reply:
Doesn't this then mean, that every test will have to use mocks, taking away the possibility to use the real implementation in only some tests?

Also some tests only need a stub and can safe some memory, others need a full spy. With one global place to define it, one can't fine tune that.